src/commands/clan/skills.ts

```diff
@@ -12,169 +12,185 @@ import {
   getRaidLog,
 } from "./_dread";
 
-const KILLMATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+defeated\D+(\d+)/;
-const SKILLMATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+used the machine/;
+const SKILL_KILL_MATCHER = /([A-Za-z0-9\-\_ ]+)\s+\(#\d+\)\s+(defeated\D+(\d+)|used the machine)/;
 
 export const data = new SlashCommandBuilder()
   .setName("skills")
   .setDescription("Get a list of everyone currently elgible for Dreadsylvania skills.");
 
-async function parseCurrentLogs(mapToUpdate: Map<string, PlayerData>) {
-  for (let clan of DREAD_CLANS) {
-    const raidLog = await getRaidLog(clan.id);
-    if (!raidLog) throw "Clan inaccessible";
-    addParticipationFromRaidLog(raidLog, mapToUpdate);
-  }
+export type Participation = { [username: string]: { skills: number; kills: number } };
+
+function mergeParticipation(a: Participation, b: Participation) {
+  return Object.entries(a).reduce(
+    (acc, [u, { skills, kills }]) => ({
+      ...acc,
+      [u]: {
+        skills: (acc[u]?.skills ?? 0) + skills,
+        kills: (acc[u]?.kills ?? 0) + kills,
+      },
+    }),
+    b
+  );
 }
 
-function addParticipantSkills(raidLog: string, mapToUpdate: Map<string, PlayerData>) {
-  const skillLines = raidLog.toLowerCase().match(new RegExp(SKILLMATCHER, "g"));
+async function getParticipationFromCurrentRaid() {
+  const raidLogs = await Promise.all(DREAD_CLANS.map((clan) => getRaidLog(clan.id)));
 
-  for (let skill of skillLines || []) {
-    const matchedSkill = skill.match(SKILLMATCHER);
-    if (!matchedSkill) continue;
+  return raidLogs.reduce(
+    (p, log) => mergeParticipation(p, getParticipationFromRaidLog(log)),
+    {} as Participation
+  );
+}
 
-    const participation = mapToUpdate.get(matchedSkill[1]);
+export function getParticipationFromRaidLog(raidLog: string) {
+  const lines = raidLog.toLowerCase().match(new RegExp(SKILL_KILL_MATCHER, "g")) || [];
```

Comment:
if you `toLowerCase()` it, you don't need the `A-Z` in the regex

but why not just use the `i` flag?

Reply:
This is there from before but I will do that